api/v1alpha2/conversion_test.go

```diff
@@ -243,3 +282,150 @@ func TestConvertMachineDeployment(t *testing.T) {
 		})
 	})
 }
+
+func TestConvert_v1alpha2_MachineSpec_To_v1alpha3_MachineSpec(t *testing.T) {
```

Comment:
Tests should be rewritten using the gomega library as shown above with other tests

Reply:
Made the change to use gomega for assertions. Please let me know if anything else is missing.

Reply:
Please remove all these new test functions. The suggestions I made above cover everything you've added here.